crates/ruff_diagnostics/src/edit.rs

```diff
@@ -1,49 +1,58 @@
-use rustpython_parser::ast::Location;
+use ruff_text_size::{TextRange, TextSize};
+
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
+use std::cmp::Ordering;
 
 /// A text edit to be applied to a source file. Inserts, deletes, or replaces
 /// content at a given location.
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct Edit {
     /// The start location of the edit.
-    location: Location,
-    /// The end location of the edit.
-    end_location: Location,
+    range: TextRange,
     /// The replacement content to insert between the start and end locations.
     content: Option<Box<str>>,
 }
 
 impl Edit {
     /// Creates an edit that deletes the content in the `start` to `end` range.
-    pub const fn deletion(start: Location, end: Location) -> Self {
+    #[inline]
+    pub const fn deletion(start: TextSize, end: TextSize) -> Self {
+        Self::range_deletion(TextRange::new(start, end))
+    }
+
+    /// Creates an edit that deletes the content in `range`.
+    pub const fn range_deletion(range: TextRange) -> Self {
         Self {
             content: None,
-            location: start,
-            end_location: end,
+            range,
         }
     }
 
     /// Creates an edit that replaces the content in the `start` to `end` range with `content`.
-    pub fn replacement(content: String, start: Location, end: Location) -> Self {
+    #[inline]
+    pub fn replacement(content: String, start: TextSize, end: TextSize) -> Self {
+        Self::range_replacement(content, TextRange::new(start, end))
+    }
+
+    /// Creates an edit that replaces the content in `range` with `content`.
+    pub fn range_replacement(content: String, range: TextRange) -> Self {
```

Comment:
I first wanted to only have `TextRange` variants because they are very convenient to create an edit that e.g. replaces a node: `Edit::range_replaced(new_text, stmt.range())`. 

But we also have many edits where we compute the range and adding `TextRange::new(...)` in the call sites made everything too verbose for my taste.

```
Edit::range("text", TextRange::new(start_offset + relative_offset, end_offset - trailing_end))
```